### PR TITLE
feat(auth): add custom OIDC IDP provider

### PR DESCRIPTION
**Pull request by bbartels**
## What does this PR do?

This PR adds the ability to add any custom OIDC IDP SSO integration. We have an air-gapped on-premise setup at work and have an internal IDP that doesn't match any of the ones currently provided.

Docs PR here: https://github.com/langfuse/langfuse-docs/pull/684

Fixes # (issue)

The below screenshot is what the login page would look like with a custom provider and `AUTH_CUSTOM_NAME: IdentityServer`

image

## Type of change
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works
There do not seem to be any existing tests around SSO and it would be rather difficult to setup. I've confirmed the changes work locally and am happy to demonstrate to a maintainer if required.

### CODE REVIEW

**Comment by marcklingen** [comment]
Thanks for your contribution on this and the docs change!

Did you try this change end-to-end with your setup?

What kind of name needs to be provided in the callback? Currently the docs change uses `cognito` which is most likely not the correct callback

**Comment by bbartels** [comment]
The `AUTH_CUSTOM_NAME` is simply the string you want to display on the custom SSO button. So if you check the provided screenshot it says `IdentityServer` as that is what `AUTH_CUSTOM_NAME` was set to in that instance.

And yes it's fully end-to-end tested with a replica scenario (I was unable to build langfuse on prem due to some prisma dependencies on the internet). I've just used the Duende IdentityServer quickstart samples to somewhat reproduce what we have on-prem. I am happy to demonstrate!

File: web/src/env.mjs [hunk 1]
```diff
@@ -71,6 +71,11 @@ export const env = createEnv({
     AUTH_COGNITO_CLIENT_SECRET: z.string().optional(),
     AUTH_COGNITO_ISSUER: z.string().url().optional(),
     AUTH_COGNITO_ALLOW_ACCOUNT_LINKING: z.enum(["true", "false"]).optional(),
+    AUTH_CUSTOM_CLIENT_ID: z.string().optional(),
+    AUTH_CUSTOM_CLIENT_SECRET: z.string().optional(),
+    AUTH_CUSTOM_ISSUER: z.string().optional(),
```
Comment (marcklingen) [inline web/src/env.mjs:76]:
```suggestion
    AUTH_CUSTOM_ISSUER: z.string().url().optional(),
```

Can we enforce url here as it is used as a hostname later on?

**Review by marcklingen — COMMENTED** [review]
This looks good overall! I've made a few minor changes and have one open question

File: web/src/env.mjs (hunk 1 above)
Comment (bbartels) [reply web/src/env.mjs:76]: Yep that should work, frankly unsure why it wasn't already aligned!

**Comment by marcklingen** [comment]
did you `pnpm i` before running this?

**Comment by marcklingen** [comment]
This looks like a build error and I cannot reproduce it locally

**Comment by bbartels** [comment]
Ah yes, I hadn't realized you made any package changes 🤦. Tested locally with your changes, works end-to-end 🙂

**Comment by marcklingen** [comment]
awesome, merging now, thanks again!

**Review by marcklingen — APPROVED** [review]

**Comment by bbartels** [comment]
Thanks for the quick review, I appreciate it!

**Pull request merged**